GoRogue.Debugger/IRoutine.cs

```diff
@@ -1,20 +1,58 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using GoRogue.GameFramework;
-using GoRogue.MapGeneration;
+using GoRogue.MapViews;
 
 namespace GoRogue.Debugger
 {
     /// <summary>
-    /// Effectively, a set of map generation rules and map transformation
-    /// rules that allow you to debug multiple concepts in the same session
+    /// Effectively, a set of map generation rules, map transformation
+    /// rules, and views of the map that allow you to debug one or more
+    /// concepts.
     /// </summary>
     public interface IRoutine
     {
-        IEnumerable<Region> Regions { get; }
+        /// <summary>
+        /// Name of the routine.
+        /// </summary>
         string Name { get; }
-        Map? BaseMap { get; }
-        Map? TransformedMap { get; }
-        Map ElapseTimeUnit();
-        Map GenerateMap();
+
+        /// <summary>
+        /// This is the map we're working with
+        /// </summary>
+        ArrayMap<TileState> Zone { get; }
```

Comment:
I think we can remove this from the interface since it's not used, and the items in `TileState` effectively correspond to multiple routines.